src/ansible_navigator/ui_framework/curses_defs.py

```diff
@@ -22,3 +23,18 @@ class CursesLinePart(NamedTuple):
 
 CursesLine = Tuple[CursesLinePart, ...]
 CursesLines = Tuple[CursesLine, ...]
+
+
+RgbTuple = Tuple[int, ...]
+
+
+class SimpleLinePart(SimpleNamespace):
```

Comment:
I think it's better to do this as a `NamedTuple` or dataclass. I'm not able to find any examples (outside of navigator) of subclassing `SimpleNamespace` and using it in this way. I think doing it this way averts all type safety:

```
[/tmp] $ cat f.py
#!/usr/bin/env python3

from types import SimpleNamespace

class Foo(SimpleNamespace):
    bar: int

hello = Foo(bar="not an int")
print(hello.bar)

[/tmp] $ mypy f.py
Success: no issues found in 1 source file
```

whereas with `NamedTuple`:

```
[/tmp] $ cat f.py
#!/usr/bin/env python3

from typing import NamedTuple

class Foo(NamedTuple):
    bar: int

hello = Foo(bar="not an int")
print(hello.bar)

[/tmp] $ mypy f.py
f.py:8: error: Argument "bar" to "Foo" has incompatible type "str"; expected "int"
Found 1 error in 1 file (checked 1 source file)
```

I think subclassing `SimpleNamespace` and introducing fields like this doesn't actually influence the behavior (or type safety) of a `SimpleNamespace`.

Reply:
good catch, a dataclass is a better pick here, since we need it to be mutable...the columns are determined after the line is built right now